Tests/SentryTests/SentryHTTPInterceptorTests.swift

```diff
@@ -0,0 +1,111 @@
+import XCTest
+
+class SentryHTTPInterceptorTests: XCTestCase {
+    
+    private static let httpUrl = "http://somedomain.com"
+    private static let httpsUrl = "https://somedomain.com"
+    private static let wsUrl = "ws://somedomain.com"
+    private static let dsnAsString = TestConstants.dsnAsString(username: "SentrySessionTrackerTests")
+        
+    private class Fixture {
+        let client: TestClient
+        let hub: TestHub
+        let scope: Scope
+        let span: Span
+        let request: URLRequest! = nil
+        let options: Options
+        
+        init() {
+            options = Options()
+            options.dsn = SentryHTTPInterceptorTests.dsnAsString
+            
+            scope = Scope()
+            client = TestClient(options: options)!
+            span = SentrySpan(context: SpanContext(operation: "SomeOperation"))
+            hub = TestHub(client: client, andScope: scope)
+            scope.span = span
+        }
+        
+        func getSut() -> SentryHttpInterceptor {
+            return SentryHttpInterceptor(request: request, cachedResponse: nil, client: nil)
+        }
+    }
+    
+    private var fixture: Fixture!
+    
+    override func setUp() {
+        super.setUp()
+        
+        fixture = Fixture()
+        SentrySDK.setCurrentHub(fixture.hub)
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        SentrySDK.setCurrentHub(nil)
+    }
+    
+    func testSessionConfiguration() {
+        let configuration = URLSessionConfiguration()
+        
+        XCTAssertNil(configuration.protocolClasses)
```

Comment:
`m`: Can we please add a test case when there are already other protocol classes.